viewer/api/models.py

```diff
@@ -24,6 +24,10 @@ def __unicode__(self):
         return u'%s, date=%s, display=%s' % (
             self.name, self.date.strftime('%Y-%m-%d'), self.display)
 
+    def get_metrics(self):
+        return (Metric.objects.filter(collection__dataset=self).distinct('id').
```

Comment:
Change this to `list(Metrics.objects....)` so it actually executes and returns a Python list rather than a Django queryset.